python/ray/workflow/workflow_storage.py

```diff
@@ -86,6 +87,95 @@ def is_recoverable(self) -> bool:
         )
 
 
+class WorkflowIndexingStorage:
+    """Access and maintance the indexing of workflow status."""
+
+    def __init__(self):
```

Comment:
Could you add some comments here inline about the protocol here and how the fault tolerance is done?

Reply:
ohhh, I forget it... let me add it